src/core/response/mod.rs

```diff
@@ -35,6 +38,51 @@ impl AuthorizationResponse {
 
         Ok(Self::Unencoded(UntypedObject(map).try_into()?))
     }
+
+    /// Validate an authorization response against a presentation definition.
+    ///
+    /// This method will parse the presentation submission from the auth response and
+    /// validate it against the provided presentation definition.
+    ///
+    /// # Parameters
+    ///
+    /// - `self` - The authorization response to validate.
+    /// - `presentation_definition` - The presentation definition to validate against.
+    ///
+    /// # Errors
+    ///
+    /// This method will return an error if the presentation submission does not match the
+    /// presentation definition.
+    ///
+    /// # Returns
+    ///
+    /// This method will return `Ok(())` if the presentation submission matches the presentation
+    /// definition.
+    pub fn validate(&self, presentation_definition: &PresentationDefinition) -> Result<()> {
+        match self {
+            AuthorizationResponse::Jwt(_jwt) => {
+                // TODO: Handle JWT Encoded authorization response.
+
+                bail!("Authorization Response Presentation Definition validation not implemented.")
+            }
+            AuthorizationResponse::Unencoded(response) => {
+                let presentation_submission = response.presentation_submission().parsed();
+
+                // Ensure the definition id matches the submission's definition id.
+                if presentation_submission.definition_id() != presentation_definition.id() {
+                    bail!("Presentation Definition ID does not match the Presentation Submission.")
+                }
+
+                // Validate the VP Token as an unencoded payload.
+                response.vp_token().validate_unencoded(
+                    presentation_definition,
+                    presentation_submission.descriptor_map(),
+                )?;
```

Comment:
This assumes that the vp_token is a W3C VCDM Verifiable Presentation. That is only one possibility. The VP token could be a base64-encoded `DocumentResponse` from ISO 18013-5/-7, an SD-JWT+KB, or a completely proprietary data format.

Validation is incredibly complex and dependent on many different contextual factors, so for now, I suggest removing this function in favour of individual methods implementing specific checks that we know we can always perform (such as the definition ID matching).

Reply:
Validation methods have been removed in commit [0e0f9](https://github.com/spruceid/oid4vp-rs/pull/8/commits/0e0f985da4e5e0827efedf86509a424f25cc4816).